pkg/credentialprovider/plugins.go

```diff
@@ -61,6 +76,10 @@ func NewDockerKeyring() DockerKeyring {
 	for _, key := range stringKeys {
 		provider := providers[key]
 		if provider.Enabled() {
```

Comment:
would it be more straightforward to check the feature gate inside the Enabled() methods of the relevant credential provider implementations?

Reply:
+1 and this would actually remove the need for `deprecatedCredentialProviders` I think

Reply:
I went ahead and moved the check there.